### ISSUE TITLE
As a powertaker or groupmember I want to see who is my groupowner

### ISSUE THREAD

**Issue by evaklopp** [buzzn/scrum#80]
**PR**

---
**Story**
As a powertaker or groupmember I want to see who is my groupowner. 

**XD**
- [x] provide person default image

InVison Mockup: https://projects.invisionapp.com/d/main#/console/10620153/268828067/preview

**DoD**

- [x] show group owner (person or org.contact person) as Ansprechpartner with foto. If no foto, use default image. No interactions when clicking. 

- [x] show Name of the groupowner. No interactions when clicking.

**Technical details**

_Frontend_

- [x] display name of group owner
- [x] if group API response has the URL for the image, display it
- [x] if not, display default image

- [x] implement display so a contact person is not required
- [x] set max-width and -height of person image to 150x150px (that's the actual image size)

excluded: 
- [ ] add owner image and name to the 403 page as well 

_Backend_

- [x] understand the current image serving code for development and non-development setups
- [x] verify if images are served correctly when storage backend is AWS
- [x] create new AWS bucket for buzzn-core
- [x] add rake task to attach all person images we already have
- [x] set up Cloudfront distribution for the new AWS bucket
- [x] set up CNAME for the distribution (`files.buzzn.io`)
- [x] cleanup: remove unused AWS buckets and distributions
- [x] API: ensure owner image url is an empty string or nil when owner doesn't have an image
- [x] configure ASSET_HOST and AWS credentials/bucket in Heroku
- [x] cleanup: reduce number of generated image sizes
- [x] ensure production and staging are set up (AWS, buckets, Cloudfront, Cloudflare, Heroku)
- [x] update Beekeeper import: set group owner as energy mentor as well
- [x] mentors API: return energy mentors instead of admin, simplify naming, cleanup naming in backend (contact, manager, mentor, owner, admin)
- [x] remove/archive old buzzn S3 buckets
- [x] ensure images update correctly when updating an existing one (invalidate Cloudfront cache)
- [x] check if the AWS configuration of the Ruby application can be simplified -- Not yet
- [x] fix image serving code for development setups
- [x] fix tests

**Comment by phillipoertel** [comment]
The bug https://github.com/buzzn/core/issues/1277 has further info about the broken "Ansprechpartner" photo.

**Comment by phillipoertel** [comment]
@evaklopp above the bug description why the user images aren't served by the application any more. It needs to be fixed before we can show the group owner's profile image in the display app.

This ticket does not include _storing_ the user images in our application, however. If we have all the owner's images, we could assign/"import" them as part of the the (beekeeper) import process, since we already import the users.

**Comment by evaklopp** [comment]
Thanks @phillipoertel for your explanations. I think we won´t have all owners-fotos at once. maybe they just drop in every now and then. But sure, we have to think about how to upload and store them. 
I also had to rewrite the Story.
I´m dividing this story now into 

- this ticket keeps the owner picture
- powersources: #83
- storing pictures: if I write a userstory for that, it would be a future optimistic version of how PhO or Wolfgang itself uploads and stores pictures, which does not fit in this release at all. I´m not sure if it is the right way to write a story for sth where I know this is not the way it should be? Honestly I don´t want to do that. So for this Display-release we need to stick with this workaround to upload pictures like you propposed it. Is it a story? 

Question: shall I/can I move the #1277 to the product backlog, what is the best way to do it?

**Comment by phillipoertel** [comment]
@evaklopp thanks for splitting power sources and this ticket. About storing the pictures: as of now we don't have any, but this story says "show Ansprechpartner with foto", so, in the scope of this ticket, we need to get them into the system somehow. No need for a separate story IMHO right now. 
I believe we can come up with a way to do that when we estimate or plan this ticket.
Let's leave #1277 where it is and use it for technical reference when needed.

**Comment by eharit** [comment]
![asset 7 3x](https://user-images.githubusercontent.com/6729674/34153676-2e46285a-e4b3-11e7-8f61-943b1d261c07.png)

**Comment by eharit** [comment]
Sorry :)

**Comment by mkristian** [comment]
it clearly looks like an owner of a localpool :)

**Comment by evaklopp** [comment]
Design accepted. ;-)

**Comment by dottorer** [comment]
@phillipoertel Looking good for Heigelstrasse (except the foto of course ;-). 

If we assign the role "GROUP_MENTOR' to the group owner on import, we should always have a default person to show at the spot. If multiple persons would have that role, we can for now show the first that comes up and later improve and maybe cycle through them in the UI in some slick fashion (@scsirdx am I dreaming here?).

If a mentor has no image file, then the default image is shown, right?

**Comment by mkristian** [comment]
@dottorer the current code uses the first manager of the group as mentor. there the code need to be adjusted to look at the role. @phillipoertel FYI https://github.com/buzzn/core/blob/master/lib/buzzn/resources/display/group_resource.rb#L14
and then it is high time to get rid of this 'managers' of a group concept.

**Comment by scsirdx** [comment]
@dottorer Current behaviour:
if mentors.length > 1, then up to 2 mentors will be displayed in app view (as before), and the first one will be displayed in 403 view.
I can add some slide animation if needed (it's an easy task), so app will display any number of mentors.

**Comment by dottorer** [comment]
@evaklopp I don't think we strictly need to show the mentor in the 403 page (your group is not enabled). This provides little additional value right now to our customers, but consumes extra work effort in this sprint. If you agree, please let's remove this DoD.

@scsirdx Maybe we hold on this item and focus on the other sprint tics.

**Comment by phillipoertel** [comment]
@evaklopp I agree with @dottorer, we'd need to allow loading the mentors, even if the whole group can't be shown.

**Comment by evaklopp** [comment]
excluded the powertaker image/name on the 403 page, thanks for the great work! Closing
